### ISSUE TITLE
Synchronize via web with authentication

### ISSUE THREAD

**Issue by maxrodro** [Mimetis/Dotmim.Sync#501]
Hello again,

I'm doing tests with the synchronization with authentication, at the server level everything seems to be fine since I did token tests with the Postman, everything is fine.

Now I made an application in Xamarin and I am doing the synchronization in this way.
https://dotmimsync.readthedocs.io/WebSecurity.html

**Error: Dotmim.Sync.Sqlite.SqliteDbMetadata has invalid vtable method slot 22 with method none**

` //Sync Data

                var token = GenerateJwtToken(AccountLogin.Name, AccountLogin.AccountId.ToString());
                HttpClient httpClient = new HttpClient();
                httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);

                // Adding the HttpClient instance to the web client orchestrator
                var serverOrchestrator = new WebClientOrchestrator("http://192.168.1.100:5001/api/sync", client: httpClient);

                // Second provider is using plain old Sql Server provider, relying on triggers and tracking tables to create the sync environment
                string dbPath = Path.Combine(Xamarin.Essentials.FileSystem.AppDataDirectory, "TestDB.db");
                var clientProvider = new SqliteSyncProvider(dbPath);

                // Creating an agent that will handle all the process
                var agent = new SyncAgent(clientProvider, serverOrchestrator);
                agent.Parameters.Add("CompanyId", companyEntity.CompanyId);

                var progress = new SynchronousProgress<ProgressArgs>(pa =>
                {
                    Console.WriteLine($"{pa.PogressPercentageString}\t {pa.Message}");
                    message = $"{pa.PogressPercentageString}\t {pa.Message}";
                });

                var s1 = await agent.SynchronizeAsync(progress);`

But I get this error without further detail.

`OnTokenValidated: {"alg":"HS256","typ":"JWT"}.{"sub":"Dotmim.Sync","email":"test@test.com","jti":"eaddbb9b-8606-44ad-b08d-03661f2e2674","http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier":"057a7d41-14fa-4175-9bdc-eeb7f581e78a","exp":1615999788,"iss":"Dotmim.Sync.Bearer","aud":"Dotmim.Sync.Bearer"}

I/dma_iassistanc( 6666): Explicit concurrent copying GC freed 2529(175KB) AllocSpace objects, 0(0B) LOS objects, 50% free, 1819KB/3MB, paused 45us total 4.021ms

/dma_iassistanc( 6666): Explicit concurrent copying GC freed 663(54KB) AllocSpace objects, 0(0B) LOS objects, 50% free, 1797KB/3MB, paused 45us total 3.557ms[0:] Type **Dotmim.Sync.Sqlite.SqliteDbMetadata has invalid vtable method slot 22 with method none**`

**Comment by Mimetis** [comment]
If you get `OnTokenValidated` when trying to sync, that means the authentication has been successful.

Does the controller works ?
I mean, do you hit the method inside the controller ?

``` cs
[Authorize]
[HttpPost]
public async Task Post()
{
 // are you able to hit this line, when debugging ?
}
```

**Comment by maxrodro** [comment]
Right, everything works fine on the server, see this debug image.

![image](https://user-images.githubusercontent.com/64550772/111351329-b0635800-8648-11eb-8b09-f138748a7115.png)

**Comment by Mimetis** [comment]
So the problem is not coming from the authentication :)

**Comment by maxrodro** [comment]
I think the same as you my friend, anything else that I can review?

**Comment by Mimetis** [comment]
No idea.
Seems your problem is coming from the sqlite database, locally.

**Comment by maxrodro** [comment]
The DB Sqlite is created.

![image](https://user-images.githubusercontent.com/64550772/111357842-4b5f3080-864f-11eb-8daa-4ea3dd7ee540.png)

I see that it includes a record, but I see that it fails before I can create the server database schema.
![image](https://user-images.githubusercontent.com/64550772/111357898-5914b600-864f-11eb-8aca-194b6da8098e.png)

There is not much information about that error on the internet, at the suggestion.

 **Dotmim.Sync.Sqlite.SqliteDbMetadata has invalid vtable method slot 22 with method none**

**Comment by maxrodro** [comment]
I did this change to this line

`var agent = new SyncAgent (clientProvider, serverOrchestrator, clientOptions, **"dbo"**);`

Now it does not present the indicated error, but now I have this topic

0%	 Session Begins.
0%	 Getting Server Schema. Scope Name:**dbo**.
0%	 Retry Sending Http Request (1/10. Waiting 500ms... )
0%	 Retry Sending Http Request (2/10. Waiting 1000ms... )
0%	 Retry Sending Http Request (3/10. Waiting 1500ms... )
0%	 Retry Sending Http Request (4/10. Waiting 2000ms... )
0%	 Retry Sending Http Request (5/10. Waiting 2500ms... )
0%	 Retry Sending Http Request (6/10. Waiting 3000ms... )
0%	 Retry Sending Http Request (7/10. Waiting 3500ms... )
0%	 Retry Sending Http Request (8/10. Waiting 4000ms... )
0%	 Retry Sending Http Request (9/10. Waiting 4500ms... )
0%	 Retry Sending Http Request (10/10. Waiting 5000ms... )
Cannot access a closed Stream.

any ideas?

**Comment by Mimetis** [comment]
Seems you are not able to access the web api from your application
`Retry Sending Http Request` happens when `DMS` is not able to send and receive an http request

**Comment by maxrodro** [comment]
Checking a bit of Dotmim code, I see that the problem occurs in this line

![image](https://user-images.githubusercontent.com/64550772/111375161-8f5c3080-8663-11eb-90da-f803c95fce4c.png)

On the red line, jump immediately to the exception, does not present a null or something in the variable

![image](https://user-images.githubusercontent.com/64550772/111375234-9edb7980-8663-11eb-9929-e1818c56c323.png)

any other idea?

**Comment by Mimetis** [comment]
Seems you did not add your scope to the web server side.
Did you add the scope to your server side ?

``` cs
services.AddSyncServer<SqlSyncChangeTrackingProvider>(connectionString, "dbo", tables);
```

**Comment by maxrodro** [comment]
![image](https://user-images.githubusercontent.com/64550772/111377149-efec6d00-8665-11eb-982c-4cad0c5f82a9.png)
 Dont Work!!!

**Comment by Mimetis** [comment]
In your `Startup.cs`, method `ConfigureServices`

**Comment by maxrodro** [comment]
![image](https://user-images.githubusercontent.com/64550772/111380195-a736b300-8669-11eb-8229-02e88a856f93.png)

look, here I have it

**Comment by maxrodro** [comment]
I solved it, I had the version (0.6.2) of Dotmin.SQLite in my Android project and in the other projects version 0.7.0, I think it was giving some kind of conflict.

**Comment by Mimetis** [comment]
ok, nice !
